### ISSUE TITLE
Rename Tags and Namespaces

### ISSUE THREAD

**Issue by JavertTheArcanine** [hydrusnetwork/hydrus#694]
I'd like the ability to select a tag/namespace (or multiple), rename it, and have it apply to the whole local tag service.

Currently, the only way I've found to accomplish this is to "open a new search page for [the tag]", highlight all of the images, remove the offending tag, and add the corrected tag. 

This is extremely tedious when trying to rename a namespace that already has 30+ tags associated with it, as I'd have to open 30+ pages for each of the tags under that namespace.

It would also be nice to be able to delete a namespace from all tags associated with that namespace, without deleting the tags themselves.

### Example 1

> `spices:cat`
> `spices:dog`
> `spices:rat`
> 
> Rename `spices:` to `species:`
> 
> `species:cat`
> `species:dog`
> `species:rat`

### Example 2

> `general:blue eyes`
> `general:black hair`
> 
> Delete `general:`
> 
> `blue eyes`
> `black hair`

### Example 3

> Rename `lore` to `character:lore (star trek)`

### Example 4

> Select `blue eyes`, `black hair` `green hair`, etc
> 
> Add prefix `appearance:` namespace
> 
> `appearance:blue eyes`
> `appearance:black hair`
> `appearance:green hair`

### Example 5

> `animal:dog`
> `creature:wendigo`
> 
> Rename namespaces to `species:`
> 
> `species:dog`
> `species:wendigo`

**Example 3** could be accomplished with siblings, but I'd like a way to make the sibling change permanent, so I could still use the original `lore` tag for something else.

**Comment by bbappserver** [comment]
Hi Javert, mass renames have caused some pretty serious problems in the past so this issue is unlikely to gain traction, but I've come up with some workarounds for your examples where applicable.

Tags can't be renamed as such, but they can be added and removed in one go which is nearly equivalent.  For local tags this is exactly the same as a rename.  For PTR tags the faulty tag will persist so long as some hash somewhere is assigned it.  For technical reasons a direct rename is undesirable, hence why you rightly mention the sibling system exists.

### Example 1
- Perform a search for the problem tag
- Select all
- Manage tags
- Remove faulty
- Add better 

### Example 2
This is resolvable with a local sibling if you don't like the display label, but I can't imagine these particular tags aren't already siblinged making the namespaced an non namespaced equivalent (typing without a namespace will produce a suggestion for the one with the namespace and they are treated as the same).

### Example 3
As you say this is also a sibling and a bad one at that.  You can't automatically assume there is only one Lore both as a name and some files contain lore that is not Lore (startrek). See `shadow -> character: shadow the hedgehog`

### Example 4
Mass creation of namespaces is discouraged, it can lead to problems like `clothing:`.

### Example 5
Local siblings do this again, this example should be resolved by ptr siblings, if it is not you can either alter it locally or in the ptr.  Again a find and replace all is many flavours of a bad idea, so each should be treated as you encounter them, or in a session where you binge after consulting the chatroom for consensus on the change.

**Comment by Mengmoshu** [comment]
@bbappserver It might interest you to know that this is an often requested feature. And if you're going to cite "technical reason" you really should spell them out. GitHub issues are very much an appropriate place for technical details, even from people who aren't developers if they're at least familiar with the code base.

Furthermore, a tag and namespace renaming system could *reduce* the problems caused on the PTR if they are a new kind of commit on the PTR, thus allowing janitors to much more clearly judge the suitability of the change.

*And* I'm pretty sure we've mentioned to you before that the presence of a workaround shouldn't be used as an argument against a feature.

**Comment by JavertTheArcanine** [comment]
@bbappserver I would like this feature more for my own _local_ tags rather than PTR, though I believe as @Mengmoshu mentioned, it could be useful there as well.

I am aware of your solution for **Example 1**, but as I mentioned above, if the offending namespace already has 30+ tags associated with it, I'd have to open 30+ pages for each of the tags under that namespace and replace them all manually. It would be far less tedious if a feature was added to facilitate this. Plus, if I changed my mind again about what I wanted the namespace to be, I wouldn't have to open 30+ pages to replace everything _again._

As for the other Examples, you raise valid concerns about how they'd interact with PTR, but a feature that handles these Examples would still be really useful for local tags.
